src/h5web/providers/my-utils.ts

```diff
@@ -104,46 +132,55 @@ export function withMyInterpretation<
   ]);
 }
 
-export function makeMyNxDataGroup(
-  name: string,
-  signal: MyHDF5Dataset,
-  opts: Opts = {}
-): MyHDF5Group {
-  return makeMyGroup(name, [signal], {
-    ...opts,
-    attributes: [
-      ...(opts.attributes || []),
-      makeStrAttr('NX_class', 'NXdata'),
-      makeStrAttr('signal', signal.name),
-    ],
-  });
-}
-
-export function makeMyNxDataGroupWithAxes(
+export function makeMyNxDataGroup<
+  U extends Record<string, MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>>
```

Comment:
I don't understand why you need a generic here

Reply:
Yeah, I had to put it back because of `AllOrNone`. The trick of doing `keyof typeof opts.axes`  was no longer working. One argument for using overrides...